sw/device/lib/aes.h

```diff
@@ -0,0 +1,95 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef _F_LIB_AES_H__
+#define _F_LIB_AES_H__
+
+#include <stddef.h>
+#include <stdint.h>
+
+/**
+ * Supported AES modes
+ */
+typedef enum aes_modes { AES_ENC = 0, AES_DEC = 1 } aes_modes_t;
+
+/**
+ * AES unit configuration options.
+ */
+typedef struct hmac_cfg {
+  /** Operational mode @see aes_modes. */
+  aes_modes_t mode;
+  /** Can be 16, 24 or 32. */
+  size_t key_len_in_bytes;
+  /** Set to 1 to only start upon getting a trigger signal. */
+  unsigned char manual_start_trigger;
+  /** Set to 1 to not stall when previous output data has not been read. */
+  unsigned char force_data_overwrite;
+} aes_cfg_t;
+
+/**
+ * Intialize AES unit to desired mode.
+ *
+ * @param aes_cfg AES configuration settings.
+ */
+void aes_init(const aes_cfg_t aes_cfg);
```

Comment:
Don't annotate non-pointer types with `const`; this doesn't actually do anything.

Reply:
Alright.